include/net/if.h

```diff
@@ -196,36 +210,37 @@ struct lifreq
+  size_t                      lifc_len;              /* Size of buffer */
   union
   {
-    FAR char                 *lifcu_buf;                /* Buffer address */
-    FAR struct lifreq        *lifcu_req;                /* Array of ifreq structures */
+    FAR char                 *lifcu_buf;             /* Buffer address */
+    FAR struct lifreq        *lifcu_req;             /* Array of ifreq structures */
   } lifc_ifcu;
 };
 
-#define lifc_buf              lifc_ifcu.lifcu_buf       /* Buffer address */
-#define lifc_req              lifc_ifcu.lifcu_req       /* Array of ifreq structures */
+#define lifc_buf              lifc_ifcu.lifcu_buf    /* Buffer address */
+#define lifc_req              lifc_ifcu.lifcu_req    /* Array of ifreq structures */
 
 /* This is the I/F request that should be used with IPv4. */
 
 struct ifreq
 {
-  char                        ifr_name[IFNAMSIZ];       /* Network device name (e.g. "eth0") */
-  int16_t                     ifr_ifindex;              /* Interface index */
+  char                         ifr_name[IFNAMSIZ];  /* Network device name (e.g. "eth0") */
+  int16_t                      ifr_ifindex;         /* Interface index */
   union
   {
-    struct sockaddr           ifru_addr;                /* IP Address */
-    struct sockaddr           ifru_dstaddr;             /* P-to-P Address */
-    struct sockaddr           ifru_broadaddr;           /* Broadcast address */
-    struct sockaddr           ifru_netmask;             /* Netmask */
-    struct sockaddr           ifru_hwaddr;              /* MAC address */
-    int                       ifru_count;               /* Number of devices */
-    int                       ifru_mtu;                 /* MTU size */
-    uint8_t                   ifru_flags;               /* Interface flags */
-    struct mii_ioctl_notify_s ifru_mii_notify;          /* PHY event notification */
-    struct mii_ioctl_data_s   ifru_mii_data;            /* MII request data */
-    struct can_ioctl_data_s   ifru_can_data;            /* CAN bitrate request data */
+    struct sockaddr            ifru_addr;           /* IP Address */
+    struct sockaddr            ifru_dstaddr;        /* P-to-P Address */
+    struct sockaddr            ifru_broadaddr;      /* Broadcast address */
+    struct sockaddr            ifru_netmask;        /* Netmask */
+    struct sockaddr            ifru_hwaddr;         /* MAC address */
+    int                        ifru_count;          /* Number of devices */
+    int                        ifru_mtu;            /* MTU size */
+    uint8_t                    ifru_flags;          /* Interface flags */
+    struct mii_ioctl_notify_s  ifru_mii_notify;     /* PHY event notification */
+    struct mii_ioctl_data_s    ifru_mii_data;       /* MII request data */
+    struct can_ioctl_data_s    ifru_can_bitrate;    /* CAN bitrate request data */
+    struct can_ioctl_filter_s  ifru_can_filter;     /* CAN filter request data */
```

Comment:
should we add the same field to lifreq?

Reply:
I think I left it out originally because there are no uses of the IPv6 struct for SocketCAN anywhere in NuttX or NuttX-apps, but the documentation comment does suggest the two structs should be identical.  Added.